tests/system/Commands/Utilities/OptimizeTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Utilities;
+
+use CodeIgniter\Test\CIUnitTestCase;
+use CodeIgniter\Test\StreamFilterTrait;
+use PHPUnit\Framework\Attributes\Group;
+
+/**
+ * @internal
+ */
+#[Group('Others')]
+final class OptimizeTest extends CIUnitTestCase
+{
+    use StreamFilterTrait;
+
+    protected function setUp(): void
+    {
+        $this->resetServices();
+
+        parent::setUp();
+    }
+
+    protected function getBuffer(): string
+    {
+        return $this->getStreamFilterBuffer();
+    }
+
+    public function testEnableConfigCaching(): void
+    {
+        command('optimize -c');
+
+        // Check if config caching is enabled
+        $this->assertFileContains('public bool $configCacheEnabled = true;', APPPATH . 'Config/Optimize.php');
+    }
+
+    public function testEnableLocatorCaching(): void
+    {
+        command('optimize -l');
+
+        // Check if locator caching is enabled
+        $this->assertFileContains('public bool $locatorCacheEnabled = true;', APPPATH . 'Config/Optimize.php');
+    }
+
+    public function testDisableCaching(): void
+    {
+        command('optimize -d');
+
+        // Check if both caches are disabled
+        $this->assertFileContains('public bool $configCacheEnabled = false;', APPPATH . 'Config/Optimize.php');
+        $this->assertFileContains('public bool $locatorCacheEnabled = false;', APPPATH . 'Config/Optimize.php');
+    }
+
+    public function testClearCache(): void
+    {
+        // Assume the function clearCache() is called within run() and we are testing its effect
+        command('optimize');
+
+        // Check if the cache file was removed
+        $this->assertFileDoesNotExist(WRITEPATH . 'cache/FactoriesCache_config');
+    }
```

Comment:
Remove this method. 
We cannot test `command('optimize')` because it removes PHPUnit.

Reply:
then how do I simulate it? how to work around this?

Reply:
Just remove this method.
I don't think this PR need to test `Check if the cache file was removed`.
If you really want to write test for it, send another PR.

Reply:
how about this command:

`command('optimize -d')`  `command('optimize -c')`  `command('optimize -l')`

Reply:
What  do you mean?
What's happened if you run these  commands?

Reply:
I'll just double-check, but I believe the behavior is the same.